assets/src/dashboard/utils/usePagePreviewSize.js

```diff
@@ -20,46 +20,102 @@ import { useEffect, useMemo, useState } from 'react';
-    descendingBreakpointKeys[0]
-  );
+const getCurrentBp = (availableContainerSpace) =>
+  descendingBreakpointKeys.reduce((current, bp) => {
+    return availableContainerSpace <= theme.breakpoint.raw[bp] ? bp : current;
+  }, descendingBreakpointKeys[0]);
 
-const sizeFromWidth = (width) => ({
+// To determine the size of a story page we take the default page size according to breakpoint
+// and then find the remaining width in the given space that the dashboard is showing stories in
+// if the container isn't important to size then respectSetWidth catches it (thumbnails or isn't a grid)
+// otherwise, we're taking the available space we have and finding out how many items in the default size we can fit in a row
+// then we calculate the grid column gutter and the page gutter
+// subtract those values from the availableContainer space to get remaining space
+// divide the remaining space by the itemsInRow
+// attach that extra space to the width
+// get height by dividing new with by PAGE_RATIO
+const sizeFromWidth = (
   width,
-  height: width / PAGE_RATIO,
-});
+  { bp, respectSetWidth, availableContainerSpace }
+) => {
+  if (respectSetWidth) {
+    return { width, height: width / PAGE_RATIO };
+  }
+  const itemsInRow = Math.floor(availableContainerSpace / width);
+  const columnGapWidth = theme.grid.columnGap[bp] * (itemsInRow - 1);
+  const pageGutter = theme.pageHorizontalGutter[bp] * 2;
+  const takenSpace = width * itemsInRow + columnGapWidth + pageGutter;
+  const remainingSpace = availableContainerSpace - takenSpace;
+  const addToWidthValue = remainingSpace / itemsInRow;
+
+  const trueWidth = width + addToWidthValue;
+  return {
+    width: trueWidth,
+    height: trueWidth / PAGE_RATIO,
+  };
+};
+
+// we want to set the size of story pages based on the available space
+// this means we need to take the window.innerWidth value and remove the built in WP nav and the dashboard nav according to breakpoints
+const getTrueInnerWidth = () => {
+  const { innerWidth } = window;
+  if (innerWidth >= theme.breakpoint.raw.tablet) {
+    return innerWidth - WP_LEFT_NAV_WIDTH - DASHBOARD_LEFT_NAV_WIDTH;
+  } else if (innerWidth < VIEWPORT_WP_LEFT_NAV_HIDES) {
+    return innerWidth;
+  } else {
+    return innerWidth - WP_LEFT_NAV_WIDTH;
+  }
+};
 
 export default function usePagePreviewSize(options = {}) {
-  const { thumbnailMode = false } = options;
-  const [bp, setBp] = useState(getCurrentBp());
+  const { thumbnailMode = false, isGrid } = options;
+  const [availableContainerSpace, setAvailableContainerSpace] = useState(
+    getTrueInnerWidth()
+  );
+
+  const [bp, setBp] = useState(getCurrentBp(availableContainerSpace));
 
   useEffect(() => {
     if (thumbnailMode) {
       return () => {};
     }
 
-    const handleResize = () => setBp(getCurrentBp());
+    const handleResize = () => setAvailableContainerSpace(getTrueInnerWidth());
+
     window.addEventListener('resize', handleResize);
```

Comment:
Let's look at debouncing this a bit, the fps is a little low right now on window resize.

Reply:
I added a debounced callback to the resize handler, it's kinda laggy now but the cut might be worth it since it only lags when resize happens?